### DISCUSSION TITLE
E3SM mapso daily output of specific variables

### DISCUSSION THREAD

**Discussion by jacob-stu-allen** [E3SM-Project/E3SM#5738] [E3SM model help]
In my E3SM runs, I need daily output of certain ocean variables (wind stress, temperature, mixed layer depth, etc.). However, when I edit the MPASO streams, I get daily output of all variables which is taking up too much space. 

Here's my edit of the streams.ocean_forward file in the src.mpaso directory of the $CASE_ROOT/SourceMods.

Any help with what I'm doing wrong? Thanks in advance!

```
  <streams>

    <stream name="timeSeriesStatsDailyOutput"
        type="output"
        filename_template="analysis_members/timeSeriesStatsDaily.$Y-$M.nc"
        filename_interval="0000-01-00_00:00:00"
        reference_time="0001-01-01_00:00:00"
        clobber_mode="truncate"
        packages="timeSeriesStatsDailyAMPKG"
        output_interval="00-01-00_00:00:00" >

        <var name="activeTracers_temperature"/>
        <var name="windStressZonal"/>
        <var name="windStressMeridional"/>
        <var name="dThreshMLD"/>
    </stream>

  </streams>
```

**Comment by jonbob** [comment]
@jacob-stu-allen -- my name is Jon Wolfe and I'm an E3SM software person working at LANL.

You did well to get the streams file modified and staged correctly in SourceMods. What exactly are you running? In E3SM the mpas-ocean streams file is called streams.ocean and not streams.ocean_forward -- so it may not be picking it up. But it would help to know more details about specifically what you're trying to do. Thanks

**Reply by jacob-stu-allen** [reply]
Hey Jon,

I'm branching a WCYCLE1850 case off of the 500 year piControl run in E3SM-2-0 and at the moment, just outputting daily data of interest from MPAS-O. I need this data to create some statistics that will be used in future simulations with some more significant SourceMod changes to the model's wind stress. 

COMPSET: WCYCL1850
RES: ne30pg2_EC30to60E2r2

Thanks,
Jacob

**Reply by jonbob** [reply]
Thanks Jason -- that helps. Please try renaming the streams file to streams.ocean, but if that's all that you have in the streams file it won't work. The streams are used for input as well as output, so what we typically do is copy the streams.ocean file from the run directory into that SourceMods spot and modify it there. But we tend to control the streams with a combination of config flags that can be changed in user_nl_mpaso and some easy alterations in the streams file. Most of the optional streams have a setting like:
````
 config_am_globalstats_enable = .true.
 ````
which turns that stream on or off. You can modify what's actually output into a stream in the streams file, but it's much  easier to simply turn a stream off then to remove it (which can cause problems). Please let me know if that makes sense

**Reply by jacob-stu-allen** [reply]
Thanks, Jon. I'm just getting to testing this out, and the process of editing the streams.ocean file is much more clear to me!
